pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -525,3 +525,12 @@ func addEndpointSubset(subsets []v1.EndpointSubset, pod *v1.Pod, epa v1.Endpoint
 	}
 	return subsets, readyEps, notReadyEps
 }
+
+func shouldPodBeInEndpoints(pod *v1.Pod) bool {
+	if (pod.Spec.RestartPolicy == v1.RestartPolicyNever && pod.Status.Phase == v1.PodFailed) ||
```

Comment:
Aha, thanks! Will do. I thought to use switch/case, but I was faint when I rebased the code:) I'd also like to write test case for `shouldPodBeInEndpoints`. Worthy?